### PR TITLE
ticdc-open-api: add ticdc open api doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
ticdc-open-api: add an file about ticdc open api

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6846
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-open-api.md [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+
+# TiCDC Open API
+
+<!-- markdownlint-disable MD024 -->
+
+> **Warning:**
+>
+> TiCDC Open API is still an experimental feature. It is not recommended to use it in a production environment.
+
+TiCDC provides the Open API feature. Using the APIs, you can query and operate the TiCDC cluster. Overall, the Open API feature is similar to that of [`cdc cli` tool] (/ticdc/manage-ticdc.md#use-cdc-cli-to-manage-cluster-status-and-data-replication-task).
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:14]:
```suggestion
TiCDC provides the Open API feature for querying and operating the TiCDC cluster. The Open API feature is similar to that of [`cdc cli` tool] (/ticdc/manage-ticdc.md#use-cdc-cli-to-manage-cluster-status-and-data-replication-task).
```

File: ticdc/ticdc-open-api.md [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+You can use the Open APIs to perform the following maintenance operations on the TiCDC cluster:
+
+- [Get the status information of a TiCDC node](#get-the-status-information-of-a-ticdc-node)
+- [Check the health status of a TiCDC cluster](#check-the-health-status-of-a-ticdc-cluster)
+- [Create a replication task](#create-a-replication-task)
+- [Remove a replication task](#remove-a-replication-task)
+- [Update the replication configuration](#update-the-replication-configuration)
+- [Query the replication task list](#query-the-replication-task-list)
+- [Query a specific replication task](#query-a-specific-replication-task)
+- [Pause a replication task] (#pause-a-replication-task)
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:25]:
```suggestion
- [Pause a replication task](#pause-a-replication-task)
```

File: ticdc/ticdc-open-api.md [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+- [Resume a replication task](#resume-a-replication-task)
+- [Query the replication subtask list](#query-the-replication-subtask-list)
+- [Query a specific replication subtask](#query-a-specific-replication-subtask)
+- [Query the TiCDC service process list](#query-the-ticdc-service-process-list)
+- [Evict an owner node](#evict-an-owner-node)
+- [Manually trigger the load balancing of a table](#manually-trigger-the-load-balancing-of-a-table)
+- [Manually schedule a table to another node](#manually-schedule-a-table-to-another-node)
+- [Dynamically adjust the log level of the TiCDC server](#dynamically-adjust-the-log-level-of-the-ticdc-server)
+
+The request body and returned value of all APIs are in JSON format. The following sections describes the specific usage of the Open APIs.
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:35]:
```suggestion
The request body and returned value of all APIs are in JSON format. The following sections describe the specific usage of the Open APIs.
```

File: ticdc/ticdc-open-api.md [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+}
+```
+
+The fields of the above output are described as follows:
+
+-version: The current TiCDC version number.
+-git_hash: The Git hash value.
+-id: The capture ID of the node.
+-pid: The capture process PID of the node.
+-is_owner: Indicates whether the node is an owner.
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:86]:
```suggestion
- version: The current TiCDC version number.
- git_hash: The Git hash value.
- id: The capture ID of the node.
- pid: The capture process PID of the node.
- is_owner: Indicates whether the node is an owner.
```

File: ticdc/ticdc-open-api.md [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+
+This is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
+
+### Request URI
+
+`POST /api/v1/changefeeds`
+
+### Parameter description
+
+The optional parameters for creating a replication task using the API are not as complete as that of using the `cli` command. This API supports the following parameters.
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:114]:
```suggestion
Compared to the optional parameters for creating a replication task using the `cli` command, the optional parameters for such task creation using the API are not as complete. This API supports the following parameters.
```

File: ticdc/ticdc-open-api.md [hunk 6]
```diff
@@ -0,0 +1,617 @@
...
+| `target_ts` | `UINT64` type. Specifies the target TSO of the changefeed. (Optional) |
+| **`sink_uri`** | `STRING` type. The downstream address of the replication task. (**Required**) |
+| `force_replicate` | `BOOLEAN` type. Determines whether to forcibly replicate the tables without unique indexes. (Optional) |
+| `ignore_ineligible_table` | `BOOLEAN` type. Determines whether to ignore the tables that cannot be replicated. (Optional) |
+| `filter_rules` | `STRING` type array. The rules for table schema filtering. (Optional) |
+| `ignore_txn_start_ts` | `UINT64` type array. Ignores the transaction of a specified start_ts. (Optional) |
+| `mounter_worker_num` | `INT` type. The mounter thread number. (Optional) |
+| `sink_config` | The configuration parameters of sink. (Optional) |
+
+The meaning and format of `changefeed_id`, `start_ts`, `target_ts`, and `sink_uri` are the same as those described in the [Use `cdc cli` to create a replication task](/ticdc/manage-ticdc.md#create-a-replication-task) document. For the detailed description of these parameters, see this document. The following section describes some other parameters in the above table.
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:131]:
```suggestion
The meaning and format of `changefeed_id`, `start_ts`, `target_ts`, and `sink_uri` are the same as those described in the [Use `cdc cli` to create a replication task](/ticdc/manage-ticdc.md#create-a-replication-task) document. For the detailed description of these parameters, see this document. The following part describes some other parameters in the above table.
```

File: ticdc/ticdc-open-api.md [hunk 7]
```diff
@@ -0,0 +1,617 @@
...
+]
+```
+
+The fields in the returned result above are described as follows:
+
+-id: The ID of the replication task.
+-state: The current [state](/ticdc/manage-ticdc.md#state-transfer-of-replication-tasks) of the replication task.
+-checkpoint_tso: The TSO representation of the current checkpoint of the replication task.
+-checkpoint_tso: The formatted time representation of the current checkpoint of the replication task.
+-error: The error information of the replication task.
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:306]:
```suggestion
- id: The ID of the replication task.
- state: The current [state](/ticdc/manage-ticdc.md#state-transfer-of-replication-tasks) of the replication task.
- checkpoint_tso: The TSO representation of the current checkpoint of the replication task.
- checkpoint_tso: The formatted time representation of the current checkpoint of the replication task.
- error: The error information of the replication task.
```

File: ticdc/ticdc-open-api.md [hunk 8]
```diff
@@ -0,0 +1,617 @@
...
+
+#### Query parameters
+
+| Parameter name | Description |
+| :------ | :---------------------------------------- ----- |
+| `state` | Optional. When this parameter is specified, the replication status information only of this state is returned. |
+
+The value options for `state` are `all`, `normal`, `stopped`, `error`, `failed`, and `finished`.
+
+If this parameter is not specified, the basic information of replication tasks in normal, stopped, and failed states is returned by default.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:265]:
```suggestion
If this parameter is not specified, the basic information of replication tasks whose state is normal, stopped, or failed is returned by default.
```

File: ticdc/ticdc-open-api.md [hunk 9]
```diff
@@ -0,0 +1,617 @@
...
+
+- id: The ID of the replication task.
+- state: The current [state](/ticdc/manage-ticdc.md#state-transfer-of-replication-tasks) of the replication task.
+- checkpoint_tso: The TSO representation of the current checkpoint of the replication task.
+- checkpoint_tso: The formatted time representation of the current checkpoint of the replication task.
+- error: The error information of the replication task.
+
+## Query a specific replication task
+
+This is a synchronous interface. If the request is successful, the detailed information of the specified replication task is returned.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:310]:
```suggestion
This API is a synchronous interface. If the request is successful, the detailed information of the specified replication task is returned.
```

File: ticdc/ticdc-open-api.md [hunk 10]
```diff
@@ -0,0 +1,617 @@
...
+            ],
+            "table_operations": {}
+        }
+    ]
+}
+```
+
+## Pause a replication task
+
+This is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:363]:
```suggestion
This API is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```

File: ticdc/ticdc-open-api.md [hunk 11]
```diff
@@ -0,0 +1,617 @@
...
+
+```shell
+curl -X POST http://127.0.0.1:8300/api/v1/changefeeds/test1/pause
+```
+
+If the request is successful, `202 Accepted` is returned. If the request fails, an error message and error code are returned.
+
+## Resume a replication task
+
+This is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:391]:
```suggestion
This API is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```

File: ticdc/ticdc-open-api.md [hunk 12]
```diff
@@ -0,0 +1,617 @@
...
+
+```shell
+curl -X POST http://127.0.0.1:8300/api/v1/changefeeds/test1/resume
+```
+
+If the request is successful, `202 Accepted` is returned. If the request fails, an error message and error code are returned.
+
+## Query the replication subtask list
+
+This is a synchronous interface. If the request is successful, the basic information of all replication subtasks is returned.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:419]:
```suggestion
This API is a synchronous interface. If the request is successful, the basic information of all replication subtasks is returned.
```

File: ticdc/ticdc-open-api.md [hunk 13]
```diff
@@ -0,0 +1,617 @@
...
+    {
+        "changefeed_id": "test1",
+        "capture_id": "561c3784-77f0-4863-ad52-65a3436db6af"
+    }
+]
+```
+
+## Query a specific replication subtask
+
+This is a synchronous interface. If the request is successful, the detailed information of the specified replication subtask (`processor`) is returned.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:444]:
```suggestion
This API is a synchronous interface. If the request is successful, the detailed information of the specified replication subtask (`processor`) is returned.
```

File: ticdc/ticdc-open-api.md [hunk 14]
```diff
@@ -0,0 +1,617 @@
...
+        63,
+        65,
+    ],
+    "error": null
+}
+```
+
+## Query the TiCDC service process list
+
+This is a synchronous interface. If the request is successful, the basic information of all replication processes (`capture`) is returned.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:482]:
```suggestion
This API is a synchronous interface. If the request is successful, the basic information of all replication processes (`capture`) is returned.
```

File: ticdc/ticdc-open-api.md [hunk 15]
```diff
@@ -0,0 +1,617 @@
...
+        "id": "561c3784-77f0-4863-ad52-65a3436db6af",
+        "is_owner": true,
+        "address": "127.0.0.1:8300"
+    }
+]
+```
+
+## Evict an owner node
+
+This is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:508]:
```suggestion
This API is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```

File: ticdc/ticdc-open-api.md [hunk 16]
```diff
@@ -0,0 +1,617 @@
...
+
+```shell
+curl -X POST http://127.0.0.1:8300/api/v1/owner/resign
+```
+
+If the request is successful, `202 Accepted` is returned. If the request fails, an error message and error code are returned.
+
+## Manually trigger the load balancing of a table
+
+This is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:528]:
```suggestion
This API is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```

File: ticdc/ticdc-open-api.md [hunk 17]
```diff
@@ -0,0 +1,617 @@
...
+
+```shell
+ curl -X POST http://127.0.0.1:8300/api/v1/changefeeds/test1/tables/rebalance_table
+```
+
+If the request is successful, `202 Accepted` is returned. If the request fails, an error message and error code are returned.
+
+## Manually schedule a table to another node
+
+This is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:556]:
```suggestion
This API is an asynchronous interface. If the request is successful, `202 Accepted` is returned. The returned result only means that the server agrees to run the command but does not guarantee that the command will be run successfully.
```

File: ticdc/ticdc-open-api.md [hunk 18]
```diff
@@ -0,0 +1,617 @@
...
+```shell
+curl -X POST -H "'Content-type':'application/json'" http://127.0.0.1:8300/api/v1/changefeeds/changefeed-test1/tables/move_table -d '{"capture_id":"6f19a6d9-0f8c-4dc9-b299-3ba7c0f216f5","table_id":49}'
+
+```
+
+If the request is successful, `202 Accepted` is returned. If the request fails, an error message and error code are returned.
+
+## Dynamically adjust the log level of the TiCDC server
+
+This is a synchronous interface. If the request is successful, `202 OK` is returned.
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:592]:
```suggestion
This API is a synchronous interface. If the request is successful, `202 OK` is returned.
```

File: ticdc/ticdc-open-api.md [hunk 19]
```diff
@@ -0,0 +1,617 @@
...
+
+### Request parameters
+
+#### Parameters for the request body
+
+| Parameter name | Description |
+| :---------- | :----------------- |
+| `log_level` | The log level you want to set. |
+
+`log_level` supports the [log levels provided by zap](https://godoc.org/go.uber.org/zap#UnmarshalText): "debug", "info", "warn", "error", "dpanic" , "Panic", "fatal".
```
Comment (Liuxiaozhen12) [inline ticdc/ticdc-open-api.md:606]:
```suggestion
`log_level` supports the [log levels provided by zap](https://godoc.org/go.uber.org/zap#UnmarshalText): "debug", "info", "warn", "error", "dpanic" , "panic", and "fatal".
```

**Review by Liuxiaozhen12 — APPROVED** [review]

File: ticdc/ticdc-open-api.md [hunk 20]
```diff
@@ -0,0 +1,617 @@
...
+- [Update the replication configuration](#update-the-replication-configuration)
+- [Query the replication task list](#query-the-replication-task-list)
+- [Query a specific replication task](#query-a-specific-replication-task)
+- [Pause a replication task](#pause-a-replication-task)
+- [Resume a replication task](#resume-a-replication-task)
+- [Query the replication subtask list](#query-the-replication-subtask-list)
+- [Query a specific replication subtask](#query-a-specific-replication-subtask)
+- [Query the TiCDC service process list](#query-the-ticdc-service-process-list)
+- [Evict an owner node](#evict-an-owner-node)
+- [Manually trigger the load balancing of a table](#manually-trigger-the-load-balancing-of-a-table)
```
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:31]: It is not a table. It is all tables in a replication task.

Comment (asddongmen) [inline ticdc/ticdc-open-api.md:31]: It is not a table. It is all tables in a replication task.

File: ticdc/ticdc-open-api.md [hunk 21]
```diff
@@ -0,0 +1,617 @@
...
+
+`PUT /api/v1/changefeeds/{changefeed_id}`
+
+### Parameter description
+
+#### Path parameters
+
+| Parameter name | Description |
+| :-------------- | :----------------------------------- |
+| `changefeed_id` | The ID of the replication task (changefeed) to be queried. |
```
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:222]:
```suggestion
| `changefeed_id` | The ID of the replication task (changefeed) to be updated. |
```

File: ticdc/ticdc-open-api.md [hunk 22]
```diff
@@ -0,0 +1,617 @@
...
+
+This is a synchronous interface. If the request is successful, the basic information of all nodes in the TiCDC cluster is returned.
+
+### Parameter description
+
+#### Query parameters
+
+| Parameter name | Description |
+| :------ | :---------------------------------------- ----- |
+| `state` | Optional. When this parameter is specified, the replication status information only of this state is returned. |
```
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:261]:
```suggestion
| `state` | When this parameter is specified, the replication status information only of this state is returned.(Optional) |
```

File: ticdc/ticdc-open-api.md [hunk 23]
```diff
@@ -0,0 +1,617 @@
...
+
+The value options for `state` are `all`, `normal`, `stopped`, `error`, `failed`, and `finished`.
+
+If this parameter is not specified, the basic information of replication tasks in normal, stopped, and failed states is returned by default.
+
+### Request URI
```
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:267]: The `### Request URI` should be right after the secondary title `## Query the replication task list`

File: ticdc/ticdc-open-api.md (hunk 12 above)
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:419]:
```suggestion
This is a synchronous interface. If the request is successful, the basic information of all replication subtasks (`processor`) is returned.
```

File: ticdc/ticdc-open-api.md [hunk 24]
```diff
@@ -0,0 +1,617 @@
...
+
+#### Path parameters
+
+| Parameter name | Description |
+| :-------------- | :----------------------------------- |
+| `changefeed_id` | The ID of the replication task (changefeed) to be scheduled. |
+
+### Example
+
+The following request triggers the load balancing of the changefeed table with the ID `test1`.
```
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:544]:
```suggestion
The following request triggers the load balancing of all tables in the changefeed with the ID `test1`.
```

File: ticdc/ticdc-open-api.md [hunk 25]
```diff
@@ -0,0 +1,617 @@
...
+
+#### Path parameters
+
+| Parameter name | Description |
+| :-------------- | :----------------------------------- |
+| `changefeed_id` | The changefeed ID of the replication subtask to be queried. |
+| `capture_id` | The capture ID of the replication subtask to be queried. |
+
+### Example
+
```
Comment (asddongmen) [inline ticdc/ticdc-open-api.md:460]: May be it is better that we add some describe about this example:

The following request query the detail information of a subtask which `changefeed_id` is `test` and `capture_id` is `561c3784-77f0-4863-ad52-65a3436db6af`. (A subtask can be indentifed by `changefeed_id` and `capture_id`).

**Comment by asddongmen** [comment]
rest LGTM

File: ticdc/ticdc-open-api.md (hunk 20 above)
Comment (TomShawn) [reply ticdc/ticdc-open-api.md:31]:
```suggestion
- [Manually trigger the load balancing of all tables in a replication task](#manually-trigger-the-load-balancing-of-all-tables-in-a-replication-task)
```

File: ticdc/ticdc-open-api.md [hunk 26]
```diff
@@ -0,0 +1,617 @@
...
+{{< copyable "shell-regular" >}}
+
+```shell
+curl -X POST http://127.0.0.1:8300/api/v1/owner/resign
+```
+
+If the request is successful, `202 Accepted` is returned. If the request fails, an error message and error code are returned.
+
+## Manually trigger the load balancing of a table
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:526]:
```suggestion
## Manually trigger the load balancing of all tables in a replication task
```

File: ticdc/ticdc-open-api.md (hunk 25 above)
Comment (TomShawn) [reply ticdc/ticdc-open-api.md:460]:
```suggestion
The following request queries the detailed information of a subtask whose `changefeed_id` is `test` and `capture_id` is `561c3784-77f0-4863-ad52-65a3436db6af`. (A subtask can be indentifed by `changefeed_id` and `capture_id`).
```

File: ticdc/ticdc-open-api.md [hunk 27]
```diff
@@ -0,0 +1,617 @@
...
+### Parameter description
+
+#### Path parameters
+
+| Parameter name | Description |
+| :-------------- | :----------------------------------- |
+| `changefeed_id` | The changefeed ID of the replication subtask to be queried. |
+| `capture_id` | The capture ID of the replication subtask to be queried. |
+
+### Example
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:459]:
```suggestion
### Example

```

File: ticdc/ticdc-open-api.md [hunk 28]
```diff
@@ -0,0 +1,617 @@
...
+The following request queries the detailed information of a subtask whose `changefeed_id` is `test` and `capture_id` is `561c3784-77f0-4863-ad52-65a3436db6af`. A subtask can be indentifed by `changefeed_id` and `capture_id`.
+{{< copyable "shell-regular" >}}
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:461]:
```suggestion

{{< copyable "shell-regular" >}}
```

File: ticdc/ticdc-open-api.md [hunk 29]
```diff
@@ -0,0 +1,617 @@
...
+
+```shell
+curl -X GET http://127.0.0.1:8300/api/v1/processors/test1/561c3784-77f0-4863-ad52-65a3436db6af
+
```
Comment (TomShawn) [inline ticdc/ticdc-open-api.md:465]:
```suggestion
```

**Review by Liuxiaozhen12 — APPROVED** [review]

File: ticdc/ticdc-open-api.md (hunk 23 above)
Comment (TomShawn) [reply ticdc/ticdc-open-api.md]: done

**Review by asddongmen — APPROVED** [review]

**Pull request merged**